### PR TITLE
change Model for cuttingPrediction

### PR DESCRIPTION
**Pull request by joBueche**
Extend models for prediction (dedicated predictions per machinenumber, when calculating for a subscription)

### CODE REVIEW

File: Applications/ProductionManager/Contracts/Predict/CuttingPredictionPart.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        public static implicit operator CuttingPredictionPart(ProductionEntityProductionOrder productionOrder)
+        {
+            var part = new CuttingPredictionPart
+            {
+                Id = productionOrder.Id,
+                Quantity = productionOrder.Quantity,
+
+                Length = productionOrder.Length,
+                Width = productionOrder.Width,
+                Thickness = productionOrder.Thickness,
```
Comment (HOMAG-Boris-Wehrle) [inline Applications/ProductionManager/Contracts/Predict/CuttingPredictionPart.cs:78]: Please clone also the Unitsystem.

File: Applications/ProductionManager/Contracts/Predict/CuttingPredictionRequest.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    /// </summary>
+    public CuttingPredictionRequest(IEnumerable<CuttingPredictionPart>? productionEntities):this()
+    {
+        ProductionEntities = productionEntities;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="CuttingPredictionRequest"/> class.
+    /// </summary>
+    public CuttingPredictionRequest(IEnumerable<CuttingPredictionPart>? productionEntities, string? machine) :this(productionEntities)
```
Comment (HOMAG-Boris-Wehrle) [inline Applications/ProductionManager/Contracts/Predict/CuttingPredictionRequest.cs:29]: "parts" might be a better name for the variable.

Comment (HOMAG-Boris-Wehrle) [inline Applications/ProductionManager/Contracts/Predict/CuttingPredictionRequest.cs:29]: "machine" vs. "machineNumber". The HOMAG-Machine-number is not known by most customers. Maybe we need to add a method which provides all known machines (number + tapio displayname).

Comment (joBueche) [reply Applications/ProductionManager/Contracts/Predict/CuttingPredictionRequest.cs:29]: The name is still from the time, we wanted to have the "real and big" productionEntity in the request, so the model "xxxPredictionPart" was meant to be replaced.
Shall I rename it everywhere incl. docu to "xxxPredictionPart"?

Comment (HOMAG-Boris-Wehrle) [reply Applications/ProductionManager/Contracts/Predict/CuttingPredictionRequest.cs:29]: yes

Comment (joBueche) [reply Applications/ProductionManager/Contracts/Predict/CuttingPredictionRequest.cs:29]: something like "subscriptionInfo". To make it really useful, we should also add the licenses to each machine, so the customer knows, which machine is attached to hich app

**Review by HOMAG-Boris-Wehrle — APPROVED** [review]

**Review by HOMAG-Boris-Wehrle — APPROVED** [review]

**Pull request merged**